frontend/src/components/ingredient/Ingredient.js

```diff
@@ -12,34 +12,34 @@ const Ingredient = ({ navigate }) => {
   const [loading, setLoading] = useState(false);
   const [unchecked, setUnchecked] = useState(true);
 
-  const FetchData = () => {
+  const getRecipes = async () => {
     setLoading(true);
-    fetch(
-      "https://westeurope.azure.data.mongodb-api.com/app/recipe_api-eixns/endpoint/recipes"
-    )
-      .then(function (response) {
-        return response.json();
-      })
-      .then(function (data) {
-        const result = JSON.parse(JSON.stringify(data));
-        setChecked(
-          checked.map((str) => {
-            return str.toLowerCase();
-          })
-        );
-        const rec = result.filter((recipe) => {
-          return searchIngredients.every((ingredient) =>
-            recipe.Ingredients.includes(ingredient)
-          );
-        });
-        setLoading(false);
-        setMatchedRecipes(rec);
-        setSearchIngredients([]);
-        setChecked([]);
-      })
-      .catch(function (err) {
-        console.log(err);
+    try {
+      const response = await fetch("/recipes/", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          targetIngredients: searchIngredients,
+        }),
       });
+      if (!response.ok) {
+        throw new Error("Network response was not ok");
+      }
+      const data = await response.json();
+      if (!Array.isArray(data.result)) {
+        throw new Error("Data is not an array");
```

Comment:
I'm a little confused as to in what circumstances this error would be invoked? Surely any call to the recipes DB is going to return an array of objects?

Reply:
I'm in the process of deleting this and cleaning out other defunct code, will commit to this PR shortly and then merge